frontend/src/pages/HomePage.tsx

```diff
@@ -32,7 +32,9 @@ export const HomePage = () => {
         setLoading(true);
         const user = await loginUser(response.idToken);
         setLoading(false);
-        user.faculty ? navigate('/chats') : navigate('/update-user');
+        user.faculty
+          ? navigate('/chats')
+          : navigate('/update-user', { state: { name: user.name } });
```

Comment:
Rather than just sending the user's name, let's send the whole user object. This is beneficial because, in the future, I think we'll also use the update user page as an edit user page (i.e. the user has already signed up but wants to change what their into).